source/opt/interface_repair_pass.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2018 Google LLC.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "interface_repair_pass.h"
+#include "source/spirv_constant.h"
+namespace spvtools {
+namespace opt {
+InterfaceRepairPass::Status InterfaceRepairPass::Process() {
+  std::unordered_map<uint32_t, Function*> id2function_;
+  for (auto& fn : *get_module()) id2function_[fn.result_id()] = &fn;
+  bool modified = false;
+  for (auto& entry : get_module()->entry_points()) {
+    std::set<uint32_t> used_variables;
+    std::function<void(Function * func)> traverser = [&](Function* func) {
+      for (auto bi = func->begin(); bi != func->end(); ++bi) {
+        for (auto ii = bi->begin(); ii != bi->end(); ++ii) {
+          if (ii->opcode() == SpvOpFunctionCall) {
+            const auto called_id = ii->GetOperand(2).words[0];
+            traverser(id2function_.at(called_id));
+          } else {
+            for (uint32_t i = 0; i < ii->NumOperands(); ++i) {
+              auto& op = ii->GetOperand(i);
+              if (op.type == SPV_OPERAND_TYPE_ID) {
```

Comment:
You can use the `Instruction::ForEachInId`.  It becomes easier to read.  Like this the person reading the code has to figure out which operands are being skipped.

Reply:
I think you missed what I meant.  I was thinking of something like this:  https://github.com/KhronosGroup/SPIRV-Tools/blob/d20c9c2cf3fb74fb6cf3738aceaa2cd50cc33633/source/opt/basic_block.cpp#L175

You changed the code to us "GetInOperand", but that will be move than just the id operands.